fe/fe-core/src/main/java/com/starrocks/analysis/FunctionCallExpr.java

```diff
@@ -472,8 +459,21 @@ public boolean isNondeterministicBuiltinFnName() {
 
     @Override
     public int hashCode() {
-        // fnParams contains all information of children Expr. No need to calculate super's hashcode again.
-        return Objects.hash(type, opcode, fnName, fnParams, nondeterministicId);
+        // @Note: fnParams is different with children Expr. use children plz.
+        return Objects.hash(super.hashCode(), type, opcode, fnName, nondeterministicId);
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (!super.equals(obj)) {
+            return false;
+        }
+        FunctionCallExpr o = (FunctionCallExpr) obj;
+        return /*opcode == o.opcode && aggOp == o.aggOp &&*/ fnName.equals(o.fnName)
+                && fnParams.isDistinct() == o.fnParams.isDistinct()
+                && fnParams.isStar() == o.fnParams.isStar()
+                && nondeterministicId.equals(o.nondeterministicId)
+                && Objects.equals(fnParams.getOrderByElements(), o.fnParams.getOrderByElements());
```

Comment:
You need to add a UT just for these two functions and one case to cover comparison between two functions with different fnParams.

Reply:
I added

Reply:
I mean the tests for functions directly NOT the test from SQL.

Reply:
Because someone would change the planner, at that time the SQL won't call this function anymore.